### PR TITLE
oshmem: replace header files in include/mpp with symlinks

### PR DESCRIPTION
**Pull request by ggouaillardet**
This is a work around to avoit what looks like a CMake bug

Thanks Paul Kapinos for the report

Fixes open-mpi/ompi#1868

(cherry picked from commit open-mpi/ompi@2a98f9fcc39fb7e5bb61c02978fdfd7abce6b768)

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by ggouaillardet** [comment]
:bot:assign: @jsquyres 
:bot:milestone:v2.0.1
:bot:label:enhancement

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1862/ for details.

File: oshmem/include/Makefile.am [hunk 1]
```diff
@@ -22,9 +24,16 @@ include_HEADERS += shmem.fh \
...
-                         mpp/shmem.fh
+# Create symlinks for the (now deprecated) files in $(includedir)/mpp directory
+install-exec-hook:
+	test -z "$(includedir)" || $(MKDIR_P) "$(DESTDIR)$(includedir)/mpp"
+	(cd $(DESTDIR)$(includedir)/mpp; rm -f shmem.h; $(LN_S) ../shmem.h shmem.h)
+	(cd $(DESTDIR)$(includedir)/mpp; rm -f shmem.fh; $(LN_S) ../shmem.fh shmem.fh)
 
+uninstall-local:
+	rm -f $(DESTDIR)/$(includedir)/mpp/shmem.h \
+	      $(DESTDIR)/$(includedir)/mpp/shmem.fh
```
Comment (jsquyres) [inline oshmem/include/Makefile.am:35]: Don't you need to remove the `mpp` directory as well?

**Comment by ggouaillardet** [comment]
@jsquyres that is an option
before writing the commit, i ran `make uninstall` and here is the content of the install dir

```
./share
./share/openmpi
./share/openmpi/amca-param-sets
./share/man
./share/man/man7
./share/man/man3
./share/man/man1
./etc
./lib
./lib/openmpi
./lib/pkgconfig
./bin
./include
./include/openmpi
./include/openmpi/ompi
./include/openmpi/ompi/mpiext
./include/openmpi/ompi/mpiext/affinity
./include/openmpi/ompi/mpiext/affinity/c
./include/openmpi/ompi/mpiext/cuda
./include/openmpi/ompi/mpiext/cuda/c
./include/openshmem
./include/openshmem/oshmem
./include/mpp
```

bottom line, directories are not removed, so for the sake of consistency, i chose not to remove the `include/mp` directory.
that is not a strong opinion, so if you feel this should be removed, i'll update the PR

**Comment by ggouaillardet** [comment]
:bot:nolabel:pushed-back

**Comment by jsquyres** [comment]
@ggouaillardet You're right; I guess I thought that `make uninstall` removed directories too, but I guess it doesn't.  👍

**Comment by jsquyres** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1893/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1930/ for details.

**Pull request merged**